Colormappable meshes

almarklein opened the pull request:
Implements #104 for meshes. The mesh material already had support of ND colormaps, so this is just a few tweaks and adding an example. Will add support for lines etc. in subsequent PR's.

* [x] Support for 1D, 2D, and 3D colormapping on Mesh. Was already implemented.
* [x] Add a check for matching map and texcoords. 
* [x] Fixed that 2-channel colors are interpreted as gray+alpha. 
* [x] Added two examples.

----
Example output: 

![image](https://user-images.githubusercontent.com/3015475/135749225-089a2f4c-300a-4614-ab8a-7534dea7a307.png)

Korijn commented:
Looks like you pushed an empty file: examples/colormap_channels.py

Korijn commented:
I'm pretty sure people will be looking for vertex coloring (and less commonly face coloring) support. E.g. distance/heat maps are often implemented with them.

almarklein commented:
Fixed the missing file. Also changed the other example to use a cylinder instead of a torus know, because it makes the demo more clear (especially the 3D colormap case).

Made issue #152 to track per-face coloring. Let's also discuss that per-vertex coloring elsewhere (FWIW we do support per-vertex coloring, but the API is a bit odd perhaps).
